### PR TITLE
Add Apex import guard

### PR DESCRIPTION
**Pull request by ericharper**

### CODE REVIEW

File: nemo/collections/nlp/modules/common/megatron/module.py [hunk 1]
```diff
@@ -34,6 +45,10 @@ class MegatronModule(torch.nn.Module):
 
     def __init__(self, share_word_embeddings=True):
         super(MegatronModule, self).__init__()
+
+        if not HAVE_APEX:
+            logging.warning("Apex was not found. Using model parallel or megatron models will error out.")
```
Comment (MaximumEntropy) [inline nemo/collections/nlp/modules/common/megatron/module.py:50]: Should we leave this as a warning or just ask the user to install Apex here?

File: nemo/collections/nlp/modules/common/megatron/module.py [hunk 2]
```diff
@@ -140,6 +155,8 @@ def float_conversion(val):
 class Float16Module(MegatronModule):
     def __init__(self, module, precision):
         super().__init__()
+        if not HAVE_APEX:
```
Comment (MaximumEntropy) [inline nemo/collections/nlp/modules/common/megatron/module.py:158]: Same as above.

**Review by MaximumEntropy — COMMENTED** [review]
Just a minor comment on warnings vs raising an error.

File: nemo/collections/nlp/modules/common/megatron/module.py [hunk 3]
```diff
@@ -15,10 +15,21 @@
 """Megatron Module"""
 
 import torch
-from apex.transformer import parallel_state, tensor_parallel
 from torch.autograd import Variable
 from torch.nn.parameter import Parameter
 
+try:
```
Comment (ekmb) [inline nemo/collections/nlp/modules/common/megatron/module.py:21]: @ericharper also in https://github.com/NVIDIA/NeMo/blob/r1.6.0/nemo/collections/nlp/modules/common/megatron/megatron_bert.py#L20

File: nemo/collections/nlp/modules/common/megatron/module.py (hunk 1 above)
Comment (ericharper) [reply nemo/collections/nlp/modules/common/megatron/module.py]: Updated the message.

File: nemo/collections/nlp/modules/common/megatron/module.py (hunk 2 above)
Comment (ericharper) [reply nemo/collections/nlp/modules/common/megatron/module.py:160]: Updated the message.

File: nemo/collections/nlp/modules/common/megatron/module.py (hunk 3 above)
Comment (ericharper) [reply nemo/collections/nlp/modules/common/megatron/module.py:21]: Was actually able to remove that file all together.

**Review by MaximumEntropy — APPROVED** [review]
LGTM!

**Pull request merged**